ArcGISRuntimeSDKQt_CppSamples/Analysis/FindServiceAreasForMultipleFacilities/FindServiceAreasForMultipleFacilities.cpp

```diff
@@ -0,0 +1,178 @@
+    return;
+
+  m_mapView = mapView;
+  m_mapView->setMap(m_map);
+
+  m_mapView->graphicsOverlays()->append(m_serviceAreasOverlay);
+
+  connect(m_facilitiesFeatureLayer, &FeatureLayer::doneLoading, this, [this](Error loadError){
+    if (!loadError.isEmpty())
+    {
+      qDebug() << loadError.message() << loadError.additionalMessage();
+      return;
+    }
+
+    if (m_facilitiesFeatureLayer->loadStatus() == LoadStatus::Loaded)
+    {
+      // zoom to the full extent of the feature layer
+      m_mapView->setViewpointGeometry(m_facilitiesFeatureLayer->fullExtent(), 100);
+    }
+  });
+
+  emit mapViewChanged();
+}
+
+void FindServiceAreasForMultipleFacilities::findServiceAreas()
+{
+  // start showing indicator
+  m_taskRunning = true;
+  emit taskRunningChanged();
+
+  m_serviceAreaTask = new ServiceAreaTask(serviceAreaTaskUrl, this);
+  m_serviceAreaTask->load();
+
+  // once service area task is done loading, create default parameters
+  connect(m_serviceAreaTask, &ServiceAreaTask::doneLoading, this, [this](Error loadError)
+  {
+    if (!loadError.isEmpty())
+    {
+      qDebug() << loadError.message() << loadError.additionalMessage();
+      return;
+    }
+    m_serviceAreaTask->createDefaultParameters();
+  });
+
+  // once default parameters created, set parameters and solve
+  connect(m_serviceAreaTask, &ServiceAreaTask::createDefaultParametersCompleted, this, [this](QUuid, ServiceAreaParameters serviceAreaParameters)
+  {
+    serviceAreaParameters.setPolygonDetail(ServiceAreaPolygonDetail::High);
+    serviceAreaParameters.setReturnPolygons(true);
+
+    QList<double> travelTimes = {1.0, 3.0};
+
+    // add service areas of 1 minute and 3 minutes travel time by car
+    serviceAreaParameters.setDefaultImpedanceCutoffs(travelTimes);
+
+    // create query parameters used to select all facilities from the feature table
+    QueryParameters queryParameters;
+    queryParameters.setWhereClause("1=1");
+
+    // add all facilities to the service area parameters
+    serviceAreaParameters.setFacilitiesWithFeatureTable(m_facilitiesTable, queryParameters);
+
+    TaskWatcher taskWatcher = m_serviceAreaTask->solveServiceArea(serviceAreaParameters);
+    if (!taskWatcher.isValid())
+      qWarning() << "Task not valid.";
+  });
+
+  connect(m_serviceAreaTask, &ServiceAreaTask::solveServiceAreaCompleted, this, [this](QUuid, ServiceAreaResult serviceAreaResult)
+  {
+    m_taskRunning = false;
+    emit taskRunningChanged();
+
+    // iterate through the facilities to get the service area polygons
+    for (int i = 0; i < serviceAreaResult.facilities().size(); ++i)
+    {
+      QList<ServiceAreaPolygon> serviceAreaPolygonList = serviceAreaResult.resultPolygons(i);
+      // create a graphic for each available polygon
+      for (int j = 0; j < serviceAreaPolygonList.size(); ++j)
+      {
+        m_serviceAreasOverlay->graphics()->append(new Graphic(serviceAreaPolygonList[j].geometry(), m_fillSymbols[j], this));
```

Comment:
Were you ever able to find out if the order of returned results is guaranteed?

If we are demonstrating best practices - it feels as if the symbol should be assigned based on the `ImpedanceCutoff`value of the `ServiceAreaPolygon`. I am not sure if it is written anywhere the list comes to us in a specific order. It might be worth reaching out to the sample author on this one.

This comment also applies to QML.